### DISCUSSION TITLE
How to copy specific folder's content from all the folders

### DISCUSSION THREAD

**Discussion by sameerthekhans** [webpack/webpack#16014] [Q&A]
I have the below folder structure and want to copy all contents from each assets folder

```
.
└── sample-app/
    └── lib/
        ├── abc/
        │   └── assets/
        │       ├── abc-example/
        │       │   └── example.png
        │       └── abc.png
        ├── pqr/
        │   └── assets/
        │       ├── pqr-example/
        │       │   └── example.png
        │       └── pqr.png
        └── xyz/
            └── assets/
                ├── xyz-example/
                │   └── example.png
                └── xyz.png
```

The output I am expecting is something like this
```
.
└── dist/
    └── assets/
        ├── abc.png
        ├── pqr.png
        ├── xyz.png
        ├── abc-example/
        │   └── example.png
        ├── pqr-example/
        │   └── example.png
        └── xyz-example/
            └── example.png
```
I am doing this in the angular project, but under the hood it uses copyWebpackPlugin, so below is the pattern I have tried which is not working
```json
{
  "glob": "**/*",
  "input": "./lib",
  "output": "assets"
}
```

If I transform this to copyWebpackPlugin syntax it would look like this
```json
{
  "from": "**/*",
  "context": "./lib",
  "to": "assets"
}
```

[Here is a link to view copyWebpackPlugin configuration in angular cli](https://github.com/angular/angular-cli/blob/dbe0dc174339d872426501c1c1dca689db2b9bad/packages/angular_devkit/build_angular/src/webpack/utils/helpers.ts#L218)

**Comment by vankop** [comment]
I think you should pass array like
```json
[
  {
    "from": "**/*",
    "context": "./lib/abc",
    "to": "assets"
  },
  {
    "from": "**/*",
    "context": "./lib/pqr",
    "to": "assets"
  },
  {
    "from": "**/*",
    "context": "./lib/xyz",
    "to": "assets"
  }
]
```

**Reply by sameerthekhans** [reply]
@vankop Thanks, This I know and its working, The problem is I have to add this multiple times, Currently I have 5 libs and it looks ugly

**Reply by vankop** [reply]
sorry, copy plugin always inherit directory structure, so you need several configurations

**Reply by sameerthekhans** [reply]
Ok If there is no way then this can be closed
